configstack/running_module.go

```diff
@@ -221,7 +221,34 @@ func (module *RunningModule) moduleFinished(moduleErr error, r *report.Report, r
 
 		if reportExperiment {
 			if err := r.EndRun(module.Module.Path); err != nil {
-				module.Logger.Errorf("Error ending run for module %s: %v", module.Module.Path, err)
+				// If the run is not found in the report, it likely means this module was an external dependency
+				// that was excluded from the queue (e.g., with --queue-exclude-external).
+				if !errors.Is(err, report.ErrRunNotFound) {
```

Comment:
I was thinking that it may be helpful to extract this logic as a separate function

Reply:
Ya, you're right.